handler.go

```diff
@@ -72,10 +72,8 @@ func (ca *Cagent) Run(outputFile *os.File, interrupt chan struct{}) {
 				retries++
 				if retries > ca.Config.OnHTTP5xxRetries {
 					log.Error("Run: hub connection error. giving up")
-					return
+					retries = 0
```

Comment:
Sorry, what is the reason to have `retries` if it's zerod here? And error tells that it's giving up, but it will continue to rerun, why is it done so?

Also, in line 69, is it expected that in case of `ErrHubTooManyRequests` it will rerun forever, even when a signal was received in `interrupt` channel?

Reply:
> This PR contains 292 changed files...Please do not include vendor to it.

We version the vendor folder for this project. Please dont review the content of vendor folder.

Reply:
> Sorry, what is the reason to have retries if it's zerod here? And error tells that it's giving up, but it will continue to rerun, why is it done so?

The reasoning is laid out in issue DEV-1668.

```
Implement retries if cagent or csender gets 5xx from the hub.
Use a default config like

on_http_5xx_retries = 4
on_http_5xx_retry_interval = 2.0

Failed connections are logged with severity info. Only if all retries have failed, an error is logged.
```

Reply:
> Please rewrite logic with retries, it seems it wouldn't work.

Please explain this further.

Reply:
> And error tells that it's giving up, but it will continue to rerun, why is it done so?

RunHeartbeat is a goroutine that runs for the duration of the server.

Reply:
> This PR contains 292 changed files...Please do not include vendor to it.

vendor change was due to virustotal match discussed elsewhere. Since no success, it is reverted.